docs/packaging-an-application/components-and-containers.md

```diff
@@ -177,20 +177,34 @@ We can use the ports property to expose a container's port (private_port) and bi
 ```
 
 ## Volumes
-We can also specify volumes that will be bind mounted from our host to our container.
+We can also specify volumes that will be mounted.
+
+Volumes are required for any persistent data created by your application. If you have data in a container that needs to be available to new versions of your app, or data that should be backed up then you will define a volume to store it. Volumes are also useful for services that require a fast filesystem such as database or cache applications.
+
+You need to specify only the `host_path` and `container_path` of the volume. When new versions of your container are deployed, the volume will be mounted in the updated container.
+
+Named Volumes: You may create a "named" volume by providing a host_path without a leading "/" (ex. `host_path: dbdata`). On creation, named volumes will copy the information inside the container_path into the host_path location. After the initial sync the volume will act as a shared folder between your host and your docker container. Only folders can be named volumes.
+
+Bind-Mount Host Volumes: If you would like to have the volume mounted at a specific location on the host then you will provide a host_path value with a leading "/" (ex. `host_path: /dbdata`). Bind-mount volumes will initially copy the host_path contents into the container_path location. After the initial sync the volume will act as a shared folder between your host and your docker container. Folders or files can be bind-mount host volumes.
```

Comment:
I left the Bind term in there because on a docker inspect that type of mount shows as `type: bind`.

As for the docs: https://docs.docker.com/engine/tutorials/dockervolumes/